tests/rptest/services/kgo_verifier_services.py

```diff
@@ -598,8 +598,11 @@ def wait_node(self, node, timeout_sec=None):
             # the same topic, or that Redpanda showed a buggy behavior with
             # idempotency: producer records should always land at the next offset
             # after the last record they wrote.
-            raise RuntimeError(
-                f"{self.who_am_i()} possible idempotency bug: {self._status}")
+            msg = f"{self.who_am_i()} possible idempotency bug: {self._status}"
+            if self._tolerate_data_loss:
+                self._redpanda.logger.warn(msg)
```

Comment:
perhaps we should reword the msg in this branch,

observed data loss instead of "possible idempotency bug"